src/main/java/org/beanpod/switchboard/dao/ChannelDaoImpl.java

```diff
@@ -40,36 +48,35 @@ public Long deleteChannel(Long id) {
     return channelRepository.deleteChannelEntitiesById(id);
   }
 
-  public ChannelDto save(ChannelDto channelDto) {
-    return channelMapper.toChannelDto(
-        channelRepository.save(channelMapper.toChannelEntity(channelDto)));
-  }
+  // InputChannelEntity ------------------------------------------------------------------------
 
   public InputChannelDto saveInputChannel(InputChannelDto inputChannelDto) {
     return inputChannelMapper.toInputChannelDto(
         inputChannelRepository.save(inputChannelMapper.toInputChannelEntity(inputChannelDto)));
   }
 
-  public OutputChannelDto saveOutputChannel(OutputChannelDto outputChannelDto) {
-    return outputChannelMapper.toOutputChannelDto(
-        outputChannelRepository.save(outputChannelMapper.toOutputChannelEntity(outputChannelDto)));
-  }
-
   public InputChannelDto getInputChannelById(Long id) {
     InputChannelEntity inputChannelEntity = inputChannelRepository.getOne(id);
     return inputChannelMapper.toInputChannelDto(inputChannelEntity);
   }
 
+  public Long deleteOutputChannelById(UserEntity user, Long id) {
+    return outputChannelRepository.deleteOutputChannelEntitiesByEncoderDeviceUserAndId(user, id);
+  }
+
+  // OutputChannelEntity -----------------------------------------------------------------------
```

Comment:
I kept this minimal comment because it helps with readability. As you suggested in your last review, it may be worthwhile to refactor. I agree with you that the class itself has the smell of a god class, but this is not new and probably should have been caught in an earlier review. As much as I would love to perfect everything, the area of responsibility of this PR was becoming too big.